File: risorse/Esempio1.java
import java.lang.String;

class Esempio1 {
	public static void fun(String x) {
		x = "Hello world";
	}
	public static void main(String[] args) {
		String s = "hello moon";
		fun(s);
		System.out.println(s);
	}
}

File: risorse/Esempio1.cpp
#include <iostream>
#include <string>
using namespace std;
void fun(string* x) {
    x = new string{"hello world"};
}
int main() {
    string* s = new string{"hello moon"};
    fun(s);
    cout << s << endl;
    // we ignore the leak due to s
}
